src/test/java/com/exasol/spark/s3/AbstractDataReadingIntegrationTest.java

```diff
@@ -0,0 +1,136 @@
+package com.exasol.spark.s3;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import org.apache.spark.api.java.function.*;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.exasol.dbbuilder.dialects.Table;
+
+/**
+ * Abstract setup for data reading integration tests.
+ */
+abstract class AbstractDataReadingIntegrationTest extends S3IntegrationTestSetup {
```

Comment:
Maybe we could extract the map functions and give them some nice names, see branch [feature/#149-proposals_ck](https://github.com/exasol/spark-connector/tree/feature/%23149-proposals_ck).